### PR TITLE
Modernize plugin to 4.0.0 — security, REST, block editor, wp-env, tests

### PR DESCRIPTION
**Pull request by dd32**
## Summary

Full modernization of the plugin to `4.0.0`. Rewrites the structure, admin UI and dev tooling from scratch. Requires WordPress 6.4+ and PHP 7.4+.

## Architecture

- PSR-4 autoloaded `src/` tree with strict types throughout:
  - `Plugin` — bootstrap / hook registration / upgrade routine
  - `Admin` — admin page container
  - `Filesystem` — safe path resolution (decoupled from WP globals for testability)
  - `Importer` — copy-and-attach pipeline
  - `RestController` — `/wp-json/add-from-server/v1/*` endpoints
  - `BlockEditor` — block editor sidebar plugin registration
- Legacy files deleted: `class.add-from-server.php`, `compat.php`, `old-versions.php`, the jQuery admin script and the old CSS.

## Simplified configuration

- The legacy `frmsvr_root` option is deleted on upgrade (`Plugin::maybe_run_upgrade`).
- Root directory is now controlled exclusively by the `ADD_FROM_SERVER` constant or the new `add_from_server_root` filter.
- The translation nag screen and the old outdated-options notice are gone.

## Block editor integration

- `assets/editor.js` registers a `PluginSidebar` + `PluginSidebarMoreMenuItem` with a modal file browser.
- Selected files import via REST and the resulting attachment is inserted as a `core/image` block.
- Admin page is now a React app using `@wordpress/components` against the same REST endpoints.
- Written with `wp.element.createElement` so **no JSX/bundler step is required** — the plugin works directly from source.

## Local environment

- `.wp-env.json` boots WordPress trunk on PHP 8.2 with the plugin mounted.
- `package.json` exposes `npm start` / `npm run test:php` / `npm run lint:php`.

## Tests

PHPUnit with separate unit and integration suites. 16 unit tests, 33 assertions, all passing:

- `tests/Unit/FilesystemTest.php` — normalization, canonical root, traversal rejection, absolute-path rejection, symlink-escape rejection, segment-boundary edge cases (e.g. `/var/wwwsomething` is correctly outside `/var/www`), directory listing.
- `tests/Unit/ImporterTest.php` — capability guard, root enforcement, missing file, disallowed filetype.
- `tests/Integration/RestControllerTest.php` — capability/traversal behavior under a real WP environment via `@wordpress/env`.

Unit tests run without booting WP (via `tests/stubs.php`), so they work on any PHP container.

## CI

- `.github/workflows/lint.yml` — PHP 7.4 → 8.3 syntax matrix + PHPCS on 8.2.
- `.github/workflows/test.yml` (new) — unit tests across 7.4 → 8.3 plus integration tests under wp-env.

## Breaking changes

- Requires WP 6.4+ and PHP 7.4+.
- `frmsvr_root` option removed; use the `ADD_FROM_SERVER` constant or `add_from_server_root` filter.

## Test plan

- [ ] `composer install && ./vendor/bin/phpunit --testsuite=unit` passes
- [ ] `./vendor/bin/phpcs` reports 0 errors
- [ ] `npm install && npm start` boots wp-env successfully
- [ ] `npm run test:php -- --testsuite=integration` passes under wp-env
- [ ] Manual: Media → Add From Server loads, lists files, imports a file
- [ ] Manual: Block editor sidebar opens, picker imports a file, image is inserted as `core/image`

https://claude.ai/code/session_01JR2iJPhuCnhz2cyE1BUKWd

### CODE REVIEW

File: add-from-server.php [hunk 1]
```diff
@@ -1,38 +1,77 @@
...
-const MIN_WP  = '5.4';
-const MIN_PHP = '7.0';
-const VERSION = '3.4.5';
+defined( 'ABSPATH' ) || exit;
 
-// Dynamic constants must be define()'d.
-define( __NAMESPACE__ . '\PLUGIN', plugin_basename( __FILE__ ) );
+const VERSION = '4.0.0';
+const MIN_WP  = '6.4';
+const MIN_PHP = '7.4';
```
[the remark this answers is not in the document]
Comment (dd32) [reply add-from-server.php:26]: WordPress provides the polyfill.
